Bottom: 2806e54b81615a443bc785d5ebd435a6ba66c82b
Top:    f510b7d3a1caaff319b2471b1481c9e2186e1646
Author: Ian Kent <raven@themaw.net>
Date:   2013-06-07 17:16:46 +0800

autofs-5.0.7 - fix fix map entry duplicate offset detection

Map entry duplicate detection was still broken.
This hopefully will fix it, at least the Conectathon duplicate offset entry
tests pass now.


---

diff --git a/lib/cache.c b/lib/cache.c
index 1e05a99..ecace4a 100644
--- a/lib/cache.c
+++ b/lib/cache.c
@@ -659,7 +659,7 @@ int cache_update_offset(struct mapent_cache *mc, const char *mkey, const char *k
 
 	me = cache_lookup_distinct(mc, key);
 	if (me && me->age == age) {
-		if (me->multi == owner)
+		if (me == owner || strcmp(me->key, key) == 0)
 			return CHE_DUPLICATE;
 	}
